components/notification-sender-config/org.wso2.carbon.identity.notification.sender.tenant.config/src/main/java/org/wso2/carbon/identity/notification/sender/tenant/config/NotificationSenderManagementServiceImpl.java

```diff
@@ -213,6 +223,29 @@ public SMSSenderDTO getSMSSender(String senderName) throws NotificationSenderMan
         return buildSmsSenderFromResource(resource);
     }
 
+    @Override
+    public SMSSenderDTO getSMSSender(String senderName) throws NotificationSenderManagementException {
+
+        try {
+            Optional<Resource> resourceOptional = getPublisherResource(senderName);
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            if (!OrganizationManagementUtil.isOrganization(tenantDomain) && !resourceOptional.isPresent()) {
```

Comment:
We can have a boolean variable for isOrganization, and use it at L232 & L235 instead of calling twice.

Reply:
```
Optional<Resource> resourceOptional = getPublisherResource(senderName);
if (resourceOptional. isPresent()) {
    return buildSmsSenderFromResource(resourceOptional.get());
}

String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
    resourceOptional = getPublisherResource(getPrimaryTenantId(tenantDomain), senderName);
    if (resourceOptional.isPresent()) {
         return buildSmsSenderFromResource(resourceOptional.get());
    }
}

throw new NotificationSenderManagementClientException(ERROR_CODE_PUBLISHER_NOT_EXISTS, senderName);
```

Reply:
Just suggested an alternative approach to organize the logic. Check and apply if it is fine.

Reply:
Fixed